pkg/sources/forward/data_forward.go

```diff
@@ -374,7 +372,7 @@ func (isdf *DataForward) forwardAChunk(ctx context.Context) {
 
 	// when we apply transformer, we don't handle partial errors (it's either non or all, non will return early),
 	// so we should be able to ack all the readOffsets including data messages and control messages
-	err = isdf.ackFromBuffer(ctx, readOffsets)
+	err = isdf.reader.Ack(ctx, readOffsets)[0]
```

Comment:
Even this theory is correct since we do all success or all failure, this line (only checking the first item) is confusing.

If we want to do that, it would be better do it in the separate function (`ackFromBuffer`), and add some comments.